src/NeoSharp.BinarySerialization/Cache/BinarySerializerCache.cs

```diff
@@ -127,14 +142,19 @@ public BinarySerializerCache(Type type)
         /// <param name="settings">Settings</param>
         public int Serialize(IBinarySerializer serializer, BinaryWriter bw, object obj, BinarySerializerSettings settings = null)
         {
+            if (_serializer != null)
+            {
+                return _serializer.Serialize(serializer, bw, obj, settings);
+            }
+
             var ret = 0;
-            var haveFilter = settings != null && settings.Filter != null; 
+            var haveFilter = settings != null && settings.Filter != null;
 
             foreach (BinarySerializerCacheEntry e in _entries)
             {
-                if (haveFilter && !settings.Filter.Invoke(e.Context)) continue;
+                if (haveFilter && !settings.Filter.Invoke(e.Context.Order)) continue;
```

Comment:
What if the order will change?

Reply:
Is a readonly field, should not change. It is used for allow to exclude some data, like signature or hash (for compute their hash or signature)